newIDE/app/src/EventsSheet/InstructionEditor/EventTextDialog.js

```diff
@@ -0,0 +1,240 @@
+// @flow
+import { Trans } from '@lingui/macro';
+import { t } from '@lingui/macro';
+
+import * as React from 'react';
+import Dialog from '../../UI/Dialog';
+import FlatButton from '../../UI/FlatButton';
+import HelpButton from '../../UI/HelpButton';
+import { Line, Column } from '../../UI/Grid';
+import ColorPicker from '../../UI/ColorField/ColorPicker';
+import MiniToolbar, { MiniToolbarText } from '../../UI/MiniToolbar';
+import SemiControlledTextField from '../../UI/SemiControlledTextField';
+
+const gd = global.gd;
+
+const toolbarItemStyle = {
+  marginRight: 10,
+};
+
+const styles = {
+  sizeTextField: {
+    width: 90,
+    ...toolbarItemStyle,
+  },
+  toolbarItem: toolbarItemStyle,
+  checkbox: toolbarItemStyle,
+};
+type RGBColor = {|
+  r?: number,
+  g?: number,
+  b?: number,
+  a?: number,
+|};
+
+type Props = {|
+  event: gdBaseEvent,
+  onClose: () => void,
+  onApply: () => void,
+|};
+
+type State = {|
+  textValue: string,
+  textColor: RGBColor,
+  backgroundColor: RGBColor,
+|};
+
+export default class EventTextDialog extends React.Component<Props, State> {
+  _applyChangesOnEvent = () => {
+    const { event } = this.props;
+    const { textValue, textColor, backgroundColor } = this.state;
+    const eventType = event.getType();
+
+    if (eventType === 'BuiltinCommonInstructions::Comment') {
+      //Text value
+      gd.asCommentEvent(event).setComment(textValue);
+
+      //Text color
+      gd.asCommentEvent(event).setTextColor(
+        parseInt(textColor.r),
+        parseInt(textColor.g),
+        parseInt(textColor.b)
+      );
+      //Background color
+      gd.asCommentEvent(event).setBackgroundColor(
+        parseInt(backgroundColor.r),
+        parseInt(backgroundColor.g),
+        parseInt(backgroundColor.b)
+      );
+    } else if (eventType === 'BuiltinCommonInstructions::Group') {
+      //Text value
+      gd.asGroupEvent(event).setName(textValue);
+
+      //Text color for group not supported in Core, instead GroupEvent.js handle this
+      //Background color
+      gd.asGroupEvent(event).setBackgroundColor(
+        parseInt(backgroundColor.r),
+        parseInt(backgroundColor.g),
+        parseInt(backgroundColor.b)
+      );
+      return;
+    }
```

Comment:
Just by removing the optional properties in RGBColor type (see my previous comment), this should solve all the flow errors that you had in this function.

Why? Because you are using the r/g/b properties directly, so if these properties are marked as optional, Flow is warning you: "You're passing a varaible that is maybe undefined to parseInt, which expect a number! Beware there is a bug hidden here!!".

If properties are not marked as optional, Flow will understand that you're passing something that is guaranteed to be a number to parseInt - si it's fine.